src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -133,7 +133,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Throw ExceptionUtilities.UnexpectedValue(Me.MemberSymbol.Kind)
             End Select
 
-            Return boundInitializer
+            Dim expressionInitializer = TryCast(boundInitializer, BoundExpression)
+            Return If(expressionInitializer IsNot Nothing AndAlso expressionInitializer.Kind <> BoundKind.BadExpression, New BoundEqualsValue(initializer, expressionInitializer), boundInitializer)
```

Comment:
@JohnHamby It feels suspicious that BoundEqualsValue node flows to flow analysis only for bad expressions. Perhaps the only flow analysis test we have for default parameter value is testing bad expression. It should be easy to clone and modify that test to test non-bad expression. I can help you with that.